### ISSUE TITLE
network.preconnect [1861889]

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [arkenfox/user.js#1756]
should we?
- https://bugzilla.mozilla.org/show_bug.cgi?id=1861889
- `https://github.com/uBlockOrigin/uBlock-issues/issues/2913`
- `network.preconnect`

class, discuss!

**Comment by Thorin-Oakenpants** [comment]
@pierov - is this something we should do by default for MB?

**Comment by PieroV** [comment]
>  is this something we should do by default for MB?

Let's see how the upstream Bug evolves, for me

**Comment by Thorin-Oakenpants** [comment]
@PieroV ok, I'll try again .. the bug is web extensions, but if we're going to ship that in uBO, why not save computing power and any future potential bugs/regressions by setting the internal browser pref

**Comment by PieroV** [comment]
Yes, we could do it, but I don't know if I am the right one to explore this preference (well, at the moment I'm working on other stuff).
I've opened https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/42233 and I'll let richard triage it.

**Comment by Thorin-Oakenpants** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preconnect

**Comment by mik0l** [comment]
The bug also affects the built-in tracking protection.

**Comment by mik0l** [comment]
This bug also affects the built-in tracking protection.

**Comment by mik0l** [comment]
Open `about:url-classifier` and enter in `http://www.google-analytics.com`

URL: `http://www.google-analytics.com` is in the list of tables:
base-email-track-digest256
analytics-track-digest256

And I also see DNS requests to `google-analytics.com`

Set `network.preconnect=false`

No DNS requests to `google-analytics.com`

**Comment by mik0l** [comment]
Open `about:url-classifier` and enter `http://www.google-analytics.com`

URL: `http://www.google-analytics.com` is in the list of tables:
base-email-track-digest256
analytics-track-digest256

And I also see DNS requests to `google-analytics.com`

Set `network.preconnect=false`

No DNS requests to `google-analytics.com`

**Comment by rusty-snake** [comment]
Where is the privacy issue with DNS requests for blocked domains?

**Comment by Thorin-Oakenpants** [comment]
well ... it is a "thing"
- Title: Hide and Seek: Revisiting DNS-based User Tracking
- Date: June 2022
- link: https://ieeexplore.ieee.org/document/9797362
- PDF: https://bpb-us-e2.wpmucdn.com/faculty.sites.uci.edu/dist/5/764/files/2022/04/eurosp22.pdf

I haven't read it. I'm not super qualified to assess this issue - my thoughts were tor protects, but MB (and AF) by default doesn't have a VPN, so I lodged this issue to drink about it, and push it on pierov 😁 who then pushed it on richard 😀

**Comment by rusty-snake** [comment]
As I understand this paper (second link), the attacker who wants to track me is the operator of my DNS (LAN Admin, ISP or Cloudflare-DNS/Google-DNS/...) or someone who compromised my DNS. The operators of the sites I visit (example.com, google-analytics.com, ...) can not track me (that way).

So let assume that I do not trust my DNS and it is an actual threat in my threat model. Then my DNS can track that I first visted google.com, then example.com, followed by depressions-help.net. But it can not see that example.com uses google analytics.

TL;DR: DNS-based User Tracking is a thing. Is DNS-based User Tracking of tracking domains a thing?

**Comment by mik0l** [comment]
I see in `about:networking` that `ssl.google-analytics.com` is not blocked at all:

`ssl.google-analytics.com 443 HTTP/2 true 1 0`

**Comment by rusty-snake** [comment]
Can be some exceptions because of a broken site or behind-the-sence or from a restrictedDomain (where addons are not allowed). If you can narrow it down. Maybe uB logger or Browser DevTools.

**Comment by mik0l** [comment]
Pref `network.preconnect=true` with DoH enabled causes a connection to ssl.google-analytics.com.

`network.preconnect=false` will solve the problem.

**Comment by mik0l** [comment]
Simply put, pref `network.preconnect=true` ignores built-in tracking protection.

**Comment by mik0l** [comment]
> Can be some exceptions

I have `privacy.antitracking.enableWebcompat=false`

**Comment by mik0l** [comment]
> Can be some exceptions

`privacy.antitracking.enableWebcompat=false` preference is set.
I can't have any more exceptions.

**Comment by mik0l** [comment]
> Can be some exceptions

`privacy.antitracking.enableWebcompat=false` preference is set.
There can be no more exceptions.

**Comment by mik0l** [comment]
I can't figure out how this holey TP works.

For example, `pagead2.googlesyndication.com` is not blocked on `feedly.com`

**Comment by mik0l** [comment]
> For example, `pagead2.googlesyndication.com` is not blocked on `feedly.com`

`extensions.webcompat.enable_shims=false`

**Comment by mik0l** [comment]
> For example, `pagead2.googlesyndication.com` is not blocked on `feedly.com`

`extensions.webcompat.enable_shims=false`

Google's gone. But other trackers still appear.

**Comment by mik0l** [comment]
`network.preconnect=false`
`extensions.webcompat.enable_shims=false`

TP is leaky the trackers bypass  it anyway.

**Comment by mik0l** [comment]
```
network.preconnect
network.predictor.enabled
```
Two preferences need to be `false`, if one of them is `true`, the built-in TP will be ignored.
